main.tf

```diff
@@ -16,7 +19,49 @@ resource "google_storage_bucket" "tamr_bucket" {
   project  = var.project_id
 
   versioning {
-    enabled = false
+    enabled = var.versioning_enabled
+  }
+
+  # this rule deletes *all* versions after X days.
+  dynamic "lifecycle_rule" {
+    for_each = local.delete_enabled
```

Comment:
I thought for_each required a map or a set, while this is just a number? has this been tested?